src/main/java/frc/robot/subsystems/CDSSubsystem.java

```diff
@@ -169,56 +196,78 @@ public void stopCDSWheel() {
     }
   }
 
-  public boolean[] getSensorStatus() {
-    sensorStatuses = colorSensors.getProximities();
+  public void stopCDSBelt() {
+    // Stops only the belt
+    CDSBeltController.set(0.0);
     if (Constants.DebugMode) {
-      frontSensorProx.setNumber(sensorStatuses[2]);
-      middleSensorProx.setNumber(sensorStatuses[1]);
-      backSensorProx.setNumber(sensorStatuses[0]);
+      SmartDashboard.putNumber("CDS Belt Speed", 0.0);
     }
+  }
 
-    boolean backStatus = sensorStatuses[0] > Constants.backSensorActivation;
-    boolean middleStatus = sensorStatuses[1] > Constants.middleSensorActivation;
-    boolean frontStatus = sensorStatuses[2] > Constants.frontSensorActivation;
-    boolean[] beamBreakArray = {backStatus, middleStatus, frontStatus};
-
-    ballCount = 0;
-    for (boolean status : beamBreakArray) {
-      if (status) {
-        ballCount++;
+  public boolean[] getSensorStatus() {
+    if (currentProxCycle % cycleWait == 0) {
+      currentProxCycle = 1;
+      sensorStatuses = colorSensors.getProximities();
+      //if (Constants.DebugMode) {
+        frontSensorProx.setNumber(sensorStatuses[2]);
+        middleSensorProx.setNumber(sensorStatuses[1]);
+        backSensorProx.setNumber(sensorStatuses[0]);
+      //}
+
+      activationArray[0] = sensorStatuses[0] > Constants.backSensorActivation;
+      activationArray[1] = sensorStatuses[1] > Constants.middleSensorActivation;
+      activationArray[2] = sensorStatuses[2] > Constants.frontSensorActivation;
+
+      ballCount = 0;
+      for (boolean status : activationArray) {
+        if (status) {
+          ballCount++;
+        }
+      }
+      if (Constants.DebugMode) {
+        CDSBallCount.setNumber(ballCount);
       }
-    }
-    if (Constants.DebugMode) {
-      CDSBallCount.setNumber(ballCount);
-    }
 
-    return beamBreakArray;
+      return activationArray;
+    } else {
+      currentProxCycle++;
+      return activationArray;
+    }
```

Comment:
```suggestion
    } 
    currentProxCycle++;
    return activationArray;

```

Since we always return activationArray, it doesn't need to be part of the if-block